pytorch_lightning/plugins/training_type/deepspeed.py

```diff
@@ -857,12 +856,6 @@ def register_plugins(cls, plugin_registry: Dict) -> None:
             offload_optimizer_device="nvme",
         )
 
-    @DDPPlugin.checkpoint_io.setter
-    def checkpoint_io(self, io: Optional[CheckpointIO]) -> None:
-        if io is not None:
-            raise MisconfigurationException("DeepSpeed does not support custom `CheckpointIO` plugin.")
-        self._checkpoint_io = io
```

Comment:
keeping this would be a problem? it doesn't actually support any checkpoint_io and we don't have any isinstance check?

Reply:
> it doesn't actually support any checkpoint_io 

Can you explain why not?

Reply:
it uses deepspeed engine
https://github.com/PyTorchLightning/pytorch-lightning/blob/9da78a94bd1e9c3088930b24133be2b11f1dbcef/pytorch_lightning/plugins/training_type/deepspeed.py#L720-L739

Reply:
This hook being overridden should not mean you cannot pass a different IO instance. Perhaps this new instance provides the user with other implementations useful for their deepspeed engine subclass

Reply:
ah! good point 👍

Reply:
if they are subclassing and overriding the default implementation, then they should be overriding the checkpoint_io setter too then? because for that case DeepspeedStrategy is the base and if something is not available in base and I am introducing it in the fork, I should also enable it?

or is this how things are done in such an unusual case?

Reply:
do you think it's misleading if users set this but don't see their component being used by the strategy?

Reply:
I think it's okay. Save and load aren't used but remove is